src/server/pegasus_server_impl.cpp

```diff
@@ -793,7 +831,14 @@ void pegasus_server_impl::on_multi_get(const ::dsn::apps::multi_get_request &req
                     }
                 }
 
-                iterate_count++;
+                iteration_count++;
```

Comment:
Can we separate all these stuff about scan limitation into a new class? I think we don't to keep all these in pegasus_server_impl. Also, you can reuse the limitation logic in scan/multiget/sortkey_count, even they may be different in some behaviors. Mostly the codes are repeated. 

```cpp
class scan_limiter {
public:
  scan_limiter(int time, int size, int count) {}
  void start() { dsn_now_ns(); }
  bool valid() const {
    if(_max_time != 0 && duration() > _max_time) { return false; } 
    if(_max_count != 0 && _iterated_count > _max_count) { return false; } 
    if(_max_size != 0 && _iterated_size() > _max_size) { return false; }
    return true;
  }
  int duration() const { /*optimization on getting time*/ } 
  void add() { _iterated_size++; _iterated_count++; }

private:
  bool _exeed_limit{false};
  int _start_time, _end_time;
  int _iterated_count;
  int _iterated_size;

  const int _max_time{0}; // zero means no limit
  const int _max_size{0};
  const int _max_count{0};
};
```